internal/setup/warning_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setup
+
+import (
+	"bytes"
+	"os"
+	"strings"
+	"testing"
+
+	"sigs.k8s.io/krew/internal/environment"
+	"sigs.k8s.io/krew/internal/testutil"
+)
+
+func TestCheckSetup_firstRun(t *testing.T) {
+	paths := environment.NewPaths("/does-not-exist")
+	var buf bytes.Buffer
+	WarnIfBinDirIsNotInPATH(&buf, paths)
+	if buf.String() != "" {
+		t.Errorf("expected no warning for the first run, found: %s", buf.String())
+	}
+}
+
+func TestCheckSetup_windows(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+	const expectedOutput = "add the\n\"%USERPROFILE%\\.krew\\bin\" directory to your PATH environment variable"
+
+	os.Setenv("KREW_OS", "windows")
+	defer func() { os.Unsetenv("KREW_OS") }()
+	paths := environment.NewPaths(tempDir.Root())
+	var buf bytes.Buffer
+	WarnIfBinDirIsNotInPATH(&buf, paths)
+	if !strings.Contains(buf.String(), expectedOutput) {
+		t.Errorf("expected %q\nto contain %q", buf.String(), expectedOutput)
+	}
+}
+
+func TestCheckSetup_unix(t *testing.T) {
+	tests := []struct {
+		name   string
+		shell  string
+		output string
+	}{
+		{
+			name:   "When the shell is zsh",
+			shell:  "/bin/zsh",
+			output: "add\nthe following to your ~/.zshrc:\n\n    export PATH=\"${PATH}:${HOME}/.krew/bin\"",
+		},
+		{
+			name:   "When the shell is bash",
+			shell:  "/bin/bash",
+			output: "add\nthe following to your ~/.bash_profile or ~/.bashrc:\n\n    export PATH=\"${PATH}:${HOME}/.krew/bin\"",
```

Comment:
since it's string.Contains, can we just say `"~/.bash_profile or ~/.bashrc"` here?
and `"config.fish"` below?

Reply:
Yeah, makes sense